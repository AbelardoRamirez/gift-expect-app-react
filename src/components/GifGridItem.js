import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({ id, type, imagen, title, source, import_datetime }) => {

    return (
        <div key={id} className="col-3 animate__animated animate__fadeIn">
            <hr className="hr shadow-lg border" />
            <div className="card text-center">
                <div className="card-header">
                    T: <span className="font-weight-bolder">{type}</span> - ID: <span className="font-weight-lighter">{id}</span>
                </div>
                <div className="card-body">
                    <img src={imagen} alt={title} className="img-fluid" />
                    <h5 className="card-title">{title}</h5>
                    <p className="card-text">Source: <span className="font-weight-lighter text-monospace">{source}</span></p>
                </div>
                <div className="card-footer text-muted">
                    {import_datetime}
                </div>
            </div>
        </div>
    );
}

GifGridItem.propTypes = {
    id: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    imagen: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    source: PropTypes.string.isRequired,
    import_datetime: PropTypes.string.isRequired,
}

export default GifGridItem;