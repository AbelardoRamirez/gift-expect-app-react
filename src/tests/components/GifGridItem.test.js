import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem/>', () => {
    test('Debe de Renderizar correctamente el componente', () => {
        const obj = {
            id: 999,
            type: 'Test',
            title: 'GOW 3',
            import_datetime: new Date().getDate().toString(),
            imagen: `https://media.giphy.com/media/kl5ctZSctCbE4/giphy.gif`,
            source: 'source.net!'
        };
        const wrapper = shallow(<GifGridItem
            {...obj}
        />);
        const img = wrapper.find('img');
        // console.log(img.html());
        // console.log(img.props());
        // console.log(img.props().src);
        const titleh5 = wrapper.find('h5');
        // console.log(titleh5.html());
        // console.log(titleh5.props());

        expect(wrapper).toMatchSnapshot();
        expect(img.props().src).toBe(obj.imagen);
        expect(titleh5.props().children).toBe(obj.title);
    });

})
