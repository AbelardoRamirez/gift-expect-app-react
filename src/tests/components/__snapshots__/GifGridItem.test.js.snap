// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe de Renderizar correctamente el componente 1`] = `
<div
  className="col-3 animate__animated animate__fadeIn"
  key="999"
>
  <hr
    className="hr shadow-lg border"
  />
  <div
    className="card text-center"
  >
    <div
      className="card-header"
    >
      T: 
      <span
        className="font-weight-bolder"
      >
        Video Games
      </span>
       - ID: 
      <span
        className="font-weight-lighter"
      >
        999
      </span>
    </div>
    <div
      className="card-body"
    >
      <img
        alt="GOW 3"
        className="img-fluid"
        src="https://media.giphy.com/media/kl5ctZSctCbE4/giphy.gif"
      />
      <h5
        className="card-title"
      >
        GOW 3
      </h5>
      <p
        className="card-text"
      >
        Source: 
        <span
          className="font-weight-lighter text-monospace"
        >
          Poringa!
        </span>
      </p>
    </div>
    <div
      className="card-footer text-muted"
    >
      20
    </div>
  </div>
</div>
`;
